models/OrderModel.php

```diff
@@ -5,47 +5,523 @@
+			$course->regular_price = (float) $course->regular_price;
+			$course->sale_price    = (float) $course->sale_price;
+			$course->image         = get_the_post_thumbnail_url( $course->id );
+		}
+
+		unset( $course );
+
+		return ! empty( $courses ) ? $courses : array();
+	}
+
+	/**
+	 * Retrieve tutor customer data by user ID.
+	 *
+	 * This function fetches customer data from the 'tutor_customers' table based on
+	 * the given user ID. It utilizes a helper function from the QueryHelper class
+	 * to perform the database query.
+	 *
+	 * The function returns the customer data as an object.
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param int $user_id The ID of the user to retrieve customer data for.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return object The customer data retrieved from the database.
+	 */
+	public function get_tutor_customer_data( $user_id ) {
+		global $wpdb;
+
+		// Retrieve customer data for the given user ID from the 'tutor_customers' table.
+		$customer_data = QueryHelper::get_row( "{$wpdb->prefix}tutor_customers", array( 'user_id' => $user_id ), 'id' );
+
+		if ( empty( $customer_data ) ) {
+			return array();
+		}
+
+		$return_data = (object) array(
+			'id'       => $customer_data->id,
+			'user_id'  => $customer_data->user_id,
+			'name'     => $customer_data->billing_name,
+			'email'    => $customer_data->billing_email,
+			'phone'    => $customer_data->billing_phone,
+			'address'  => $customer_data->billing_address,
+			'city'     => $customer_data->billing_city,
+			'state'    => $customer_data->billing_state,
+			'country'  => $customer_data->billing_country,
+			'zip_code' => $customer_data->billing_zip_code,
+		);
+
+		return $return_data;
+	}
+
+	/**
+	 * Retrieve order activities by order ID.
+	 *
+	 * This function fetches all order activities from the 'tutor_ordermeta' table
+	 * based on the given order ID and the 'history' meta key. It uses a helper
+	 * function from the QueryHelper class to perform the database query.
+	 *
+	 * If no order activities are found, the function returns an empty array.
+	 * Otherwise, it decodes the JSON-encoded meta values and returns them as an array.
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param int $order_id The ID of the order to retrieve activities for.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return array An array of order activities, each decoded from its JSON representation.
+	 */
+	public function get_order_activities( $order_id ) {
+		global $wpdb;
+
+		// Retrieve order activities for the given order ID from the 'tutor_ordermeta' table.
+		$order_activities = QueryHelper::get_all(
+			"{$wpdb->prefix}tutor_ordermeta",
+			array(
+				'order_id' => $order_id,
+				'meta_key' => self::META_KEY_HISTORY,
+			),
+			'id'
+		);
+
+		if ( empty( $order_activities ) ) {
+			return array();
+		}
+
+		$response = array();
+
+		foreach ( $order_activities as &$activity ) {
```

Comment:
Don't need to use `&` here. 
```suggestion
		foreach ( $order_activities as $activity ) {
```